apps/docs/pages/guides/auth/sessions.mdx

```diff
@@ -0,0 +1,151 @@
+
+The general rule is that a refresh token can only be used once. However, strictly enforcing this rule can cause certain issues to arise. There are two exceptions to this designed to prevent the early and unexpected termination of user's sessions:
+
+- A refresh token can be used twice or more within a defined reuse interval. By default this is 10 seconds and we do not recommend changing this value. This exception is granted for legitimate situations such as:
+    - Using server-side rendering where the same refresh token needs to be reused on the server and soon after on the client
+    - To allow some leeway for bugs or issues with serializing access to the refresh token request
+- If the parent of the currently active refresh token for the user's session is being used, the active token will be returned. This exception solves an important and often occurring situation:
+    - All clients such as browsers, mobile or desktop apps, and even some servers are inherently unreliable. Just because they sent a request, does not mean they received a response or even processed the response they received.
+    - If a refresh token is used only once, and the client did not receive and process the response, when the client comes back it will be using a refresh token that was already used and falls out of the reuse interval, causing sudden and unexpected session termination.
+
+Should the reuse attempt not fall under these two exceptions, the whole session is regarded as terminated and all refresh tokens belonging to it are marked as revoked. You can switch off this behavior in the Advanced Settings of the [Auth settings](/dashboard/project/_/settings/auth) page, though it is generally not recommended.
+
+The purpose of this mechanism is to guard against potential security issues where a refresh token could have been stolen from the user, for example by exposing it accidentally in logs that leak (like logging cookies, request bodies or URL params) or via vulnerable third-party servers. It does not guard against the case where a user's session is stolen from their device.
+
+### What are the benefits of using access and refresh tokens instead of traditional sessions?
+
+Traditionally user sessions were implemented by using a unique string stored in cookies that identified the authorization that the user had on a specific browser. Applications would use this unique string to constantly fetch the attached user information on every API call.
+
+This approach has some significant technical drawbacks, as it meant that the authentication server and database was central and inseparable from the entirety of the application:
+
+- Requirement to be always available and performant. If the authentication server or its database crashed or wasn't available for even a few seconds, the whole application went down. Scheduling maintenance or dealing with transient errors often became impossible to manage.
```

Comment:
```suggestion
- Requirement to be always available and performant. If the authentication server or its database crashes or is unavailable for even a few seconds, the whole application goes down. Scheduling maintenance or dealing with transient errors becomes very difficult.
```

if it is truly absolutely impossible, then stick with that, but if not, err on the side of just very difficult 😄